app/scripts/modules/core/src/managed/ManagedResourceObject.tsx

```diff
@@ -25,19 +26,61 @@ function getIconTypeFromKind(kind: string) {
   return kindIconMap[getKindName(kind)] ?? 'placeholder';
 }
 
-export const ManagedResourceObject = ({
-  resource: {
+const resourceKindParsingRegex = /(?<provider>\w+)\/(?<resourceType>[\w-]+)\@(?<version>\w+)/;
+const resourceNameParsingRegex = /:([\w-]+)$/;
```

Comment:
Created these regular expressions with the following assumptions.

1. `resource.kind` is of the format `provider/resourceType/version`
2. `resource.id` is of the format `provider:resourceType:account:resourceName`